src/pages/exchange-crypto/exchange-crypto.ts

```diff
@@ -66,217 +103,508 @@ export class ExchangeCryptoPage {
-      this.isOpenSelectorFrom = true;
-      walletsForActionSheet = this.fromWallets;
-      selectedWalletId = this.fromWalletSelected
-        ? this.fromWalletSelected.id
-        : null;
-    } else if (selector == 'to') {
-      this.isOpenSelectorTo = true;
-      walletsForActionSheet = this.toWallets;
-      selectedWalletId = this.toWalletSelected
-        ? this.toWalletSelected.id
-        : null;
-    }
-    const params = {
+
+    this.isOpenSelectorFrom = true;
+    walletsForActionSheet = this.fromWallets;
+    selectedWalletId = this.fromWalletSelected
+      ? this.fromWalletSelected.id
+      : null;
+
+    const walletSelector = this.actionSheetProvider.createWalletSelector({
       wallets: walletsForActionSheet,
       selectedWalletId,
-      title
-    };
-    const walletSelector = this.actionSheetProvider.createWalletSelector(
-      params
-    );
+      title: this.fromWalletSelectorTitle
+    });
     walletSelector.present();
     walletSelector.onDidDismiss(wallet => {
       this.isOpenSelectorFrom = false;
       this.isOpenSelectorTo = false;
 
-      if (!_.isEmpty(wallet)) this.onWalletSelect(wallet, selector);
+      setTimeout(() => {
+        if (!_.isEmpty(wallet)) this.onFromWalletSelect(wallet);
+      }, 100);
     });
   }
 
-  private showToWallets(): void {
-    this.toWallets = this.allWallets.filter(
-      w =>
-        !w.needsBackup &&
-        w.id != this.fromWalletSelected.id &&
-        w.coin != this.fromWalletSelected.coin
+  public showToWallets(): void {
+    if (this.toWalletSelectedByDefault || !this.fromWalletSelected) return;
+    this.isOpenSelectorTo = true;
+
+    let supportedCoins: any[];
+    let showOneInchTokensSearchBtn: boolean = false;
+
+    if (
+      this.oneInchSupportedCoins.includes(this.fromWalletSelected.coin) &&
+      this.changellySupportedCoins.includes(this.fromWalletSelected.coin)
+    ) {
+      supportedCoins = _.clone(this.exchangeCryptoSupportedCoins);
+      showOneInchTokensSearchBtn = true;
+    } else if (
+      this.oneInchSupportedCoins.includes(this.fromWalletSelected.coin)
+    ) {
+      supportedCoins = _.clone(this.oneInchSupportedCoins);
+      showOneInchTokensSearchBtn = true;
+    } else if (
+      this.changellySupportedCoins.includes(this.fromWalletSelected.coin)
+    ) {
+      supportedCoins = _.clone(this.changellySupportedCoins);
+      showOneInchTokensSearchBtn = false;
+    }
+
+    const index = supportedCoins.indexOf(this.fromWalletSelected.coin);
+    if (index > -1) {
+      supportedCoins.splice(index, 1);
+    }
+
+    const oneInchAllSupportedCoins = this.oneInchAllSupportedCoins.filter(
+      token => {
+        return (
+          token.symbol.toLowerCase() != this.fromWalletSelected.coin &&
+          token.symbol.toLowerCase() != 'eth'
+        );
+      }
+    );
+
+    let modal = this.modalCtrl.create(
+      CoinAndWalletSelectorPage,
+      {
+        title: this.toWalletSelectorTitle,
```

Comment:
replace `title` for `walletSelectorTitle`

Reply:
Done